lib/manageiq/environment.rb

```diff
@@ -19,6 +19,15 @@ def self.manageiq_plugin_setup
       setup_test_environment(:task_prefix => 'app:', :root => plugin_root)
     end
 
+    def self.manageiq_plugin_update
+      # determine plugin root dir. Assume we are called from a 'bin/setup' script in the plugin root
+      plugin_root = Pathname.new(caller_locations.last.absolute_path).dirname.parent
```

Comment:
way too magic IMHO., please pass a dir argument.  bin/setup etc should be debuggable by everyone, focusing on commands that it actually runs, not ruby magic...